src/time/duration-panel.vue

```diff
@@ -0,0 +1,163 @@
+<template>
+  <div :class="`${prefixClass}-time`">
+    <div v-if="showDurationHeader" :class="`${prefixClass}-time-header`">
+      <button
+        type="button"
+        :class="`${prefixClass}-btn ${prefixClass}-btn-text ${prefixClass}-time-header-title`"
+        @click="handleClickTitle"
+      >
+        {{ title }}
+      </button>
+    </div>
+    <div :class="`${prefixClass}-time-content`">
+      <list-columns
+        :date="innerValue"
+        :get-classes="getClasses"
+        :hour-options="hourOptions"
+        :minute-options="minuteOptions"
+        :second-options="secondOptions"
+        :hour-step="hourStep"
+        :minute-step="minuteStep"
+        :second-step="secondStep"
+        v-bind="ShowHourMinuteSecondAMPM"
+        @select="handleSelect"
+      ></list-columns>
+    </div>
+  </div>
+</template>
+
+<script>
+import { format } from 'date-format-parse';
+import { getValidDate } from '../util/date';
+import ListColumns from './list-columns';
+import { getLocaleFieldValue } from '../locale';
+
+export default {
+  name: 'DurationPanel',
+  components: { ListColumns },
+  inject: {
+    t: {
+      default: () => getLocaleFieldValue,
+    },
+    prefixClass: {
+      default: 'mx',
+    },
+  },
+  props: {
+    value: {},
+    defaultValue: {
+      default() {
+        const date = new Date();
+        date.setHours(0, 0, 0, 0);
+        return date;
```

Comment:
@Xanngo newline above pls

Reply:
Following initial repo coding styles, as discussed in slack.